pkg/api/v1beta3/types.go

```diff
@@ -241,8 +241,9 @@ type PersistentVolumeSpec struct {
 	// Resources represents the actual resources of the volume
 	Capacity ResourceList `json:"capacity,omitempty" description:"a description of the persistent volume's resources and capacity"`
 	// Source represents the location and type of a volume to mount.
-	// AccessModeTypes are inferred from the Source.
 	PersistentVolumeSource `json:",inline" description:"the actual volume backing the persistent volume"`
+	// AccessModes contains all ways the volume can be mounted
+	AccessModes []AccessModeType `json:"accessModes,omitempty" description:"all ways the volume can be mounted"`
```

Comment:
These can be gleaned from the volume plugin and persisted by PersistentVolumeManager during it's normal synchronization loop.  It allows the access modes to go from the volume plugin over the wire to the client.  Added to Spec because that's where I'm storing the ClaimRef when bound to a Claim.